File: 3-data-structures/3-1-contiguous-vs-linked-ds/array.c
#include <stdio.h>

#define ARRAY_SIZE 10

int main() { 
    int n[ARRAY_SIZE];
    int i;    

    /* Initialise all elements of array to 0 and print their memory address */
    for (i = 0; i < ARRAY_SIZE; i++) {
        n[i] = 0;
        printf("Element %d has memory address %p.\n", i, &n[i]);
    } 

}

File: 3-data-structures/3-1-contiguous-vs-linked-ds/array.py
# Going to print same memory address of each element in array

ARRAY_SIZE = 10
array = [0] * ARRAY_SIZE

def print_array(array):
    for i in range(len(array)):
        print(f"Element {i} has memory address {hex(id(array[i]))}.")

print_array(array)
